src/sentry/sentry_apps/tasks/sentry_apps.py

```diff
@@ -313,10 +314,71 @@ def _process_resource_change(
                 data[name] = serialize(instance)
 
             for installation in installations:
-                # Trigger a new task for each webhook
-                send_resource_change_webhook.delay(
-                    installation_id=installation.id, event=str(event), data=data
-                )
+
+                if _is_project_allowed(installation, instance.project_id):
+                    # Trigger a new task for each webhook
+                    send_resource_change_webhook.delay(
+                        installation_id=installation.id, event=str(event), data=data
+                    )
+
+
+def _is_project_allowed(installation: RpcSentryAppInstallation, project_id: int) -> bool:
+    service_hook = _load_service_hook(installation.organization_id, installation.id)
+    if not service_hook:
+        logger.info("send_webhooks.missing_servicehook", extra={"installation_id": installation.id})
+        return False
+    # must use 2 separate helpers for cache invalidation to work correctly in [] <-> [project1, ...] scenarios
+    return not _is_project_filtering_enabled(service_hook.id) or _does_project_filter_allow_project(
+        service_hook.id, project_id
+    )
+
+
+@cache_func_for_models(
+    [
+        (
+            ServiceHook,
+            lambda service_hook: (service_hook.organization_id or -1, service_hook.actor_id or -1),
```

Comment:
not defaulting to -1 broke a bunch of unrelated tests that did not specify `organization_id` when creating test ServiceHook objects. A bit scary how something like this can break unrelated functionality. IMHO `cache_func_for_models` should not crash on `None` values

Reply:
I'm a bit confused here. If service hooks can have a null organization_id, then this function needs to be written in a way that it can handle `organization_id: int | None`. That's the cause of the failure, I'd strongly recommend not including the -1 here.

Is it actually valid that a `ServiceHook` can have a null org id, or is it just laziness in the tests?